### ISSUE TITLE
False positive in AnnotationUseStyle for version 8.17

### ISSUE THREAD

**Issue by davidburstromspotify** [checkstyle/checkstyle#6446]
https://checkstyle.org/config_annotation.html#AnnotationUseStyle
https://checkstyle.org/property_types.html#closingParens

```
/var/tmp $ javac Example.java

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN" "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="AnnotationUseStyle" />
    </module>
</module>

/var/tmp $ cat Example.java
@Anno(@Value)
public class Example {
}

@interface Anno {
    Value value();
}

@interface Value {
}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-8.17-all.jar -c config.xml Example.java
Starting audit...
[ERROR] /var/tmp/Example.java:1: Annotation cannot have closing parenthesis. [AnnotationUseStyle]
Audit done.
Checkstyle ends with 1 errors.
```
---------------

I don't expect an error, as the closing parenthesis is for the outer annotation.

--------------

**Comment by davidburstromspotify** [comment]
I should mention this is not an issue in 8.16.

**Comment by rnveach** [comment]
Violation is happening.

> I should mention this is not an issue in 8.16.

@davidburstromspotify This statement is not true for me. I get regression going all the way back to 5.0 when the check was created.

Here is 8.16:
````
$ cat TestClass.java
@Anno(@Value)
public class Example {
}

@interface Anno {
    Value value();
}

@interface Value {
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN" "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="AnnotationUseStyle" />
    </module>
</module>

$ java -jar checkstyle-8.16-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[ERROR] TestClass.java:1: Annotation cannot have closing parenthesis. [AnnotationUseStyle]
Audit done.
Checkstyle ends with 1 errors.
````

**Comment by davidburstromspotify** [comment]
@rnveach Thank you for pointing that out.

Upon further investigation, the scenario that worked in 8.16 but broke in 8.17 was
`@Anno(value = @Value)`

**Comment by romani** [comment]
@davidburstromspotify, just curious, what for such code (annotation inside annotation) might be used ?
Can you share with us some other non simple usage of annotations? We will extend our test base. I really surprised that it is possible/compilable.

**Comment by rnveach** [comment]
@romani Are both cases false positives or is it just the 2nd one?
> `@Anno(@Value)`
> `@Anno(value = @Value)`

I have a feeling this issue was created because of pitest removals.

**Comment by davidburstromspotify** [comment]
For example, this can be used to supply test data, like `@TestData(@AccessPoint)` where `@AccessPoint` has a default value for the AP, as opposed to `@TestData(@AccessPoint("127.0.01"))`

**Comment by romani** [comment]
On default config of Check:
```
@Anno(@Value) // no violation
@Anno(value = @Value) //no violation
@Anno() // violation
@Anno(@Value()) // violation
@Anno(value = @Value()) //violation
```

**Comment by davidburstromspotify** [comment]
That looks like what we want, yes.

**Comment by rnveach** [comment]
> @Anno(@Value) // no violation

This produced a violation even in 8.16 before breaking change came in.
I will move this to a new issue.

**Comment by romani** [comment]
Fix is merged
